controller/array_action/array_mediator_svc.py

```diff
@@ -842,3 +871,192 @@ def get_array_fc_wwns(self, host_name):
+        copy_type = self._get_replication_copy_type(rcrelationship)
+        is_primary = self._is_replication_endpoint_primary(rcrelationship)
+        is_ready = self._is_replication_ready(rcrelationship)
+        return Replication(name=rcrelationship.name,
+                           volume_internal_id=volume_internal_id,
+                           other_volume_internal_id=other_volume_internal_id,
+                           copy_type=copy_type,
+                           is_primary=is_primary,
+                           is_ready=is_ready)
+
+    def _get_lsrcrelationship(self, filter_value):
+        return self.client.svcinfo.lsrcrelationship(filtervalue=filter_value)
+
+    def _get_rcrelationship_by_name(self, replication_name, not_exist_error=True):
+        filter_value = 'RC_rel_name={0}'.format(replication_name)
+        rcrelationship = self._get_lsrcrelationship(filter_value).as_single_element
+        if not rcrelationship and not_exist_error:
+            raise array_errors.ObjectNotFoundError(replication_name)
+        return rcrelationship
+
+    def _get_rcrelationships(self, cli_volume_id, other_cli_volume_id, other_system_id, as_master):
+        endpoint_type = ENDPOINT_TYPE_AUX
+        other_endpoint_type = ENDPOINT_TYPE_MASTER
+        if as_master:
+            endpoint_type = ENDPOINT_TYPE_MASTER
+            other_endpoint_type = ENDPOINT_TYPE_AUX
+        filter_value = '{END}_vdisk_id={VDISK_ID}:' \
+                       '{OTHER_END}_vdisk_id={OTHER_VDISK_ID}:' \
+                       '{OTHER_END}_cluster_id={OTHER_CLUSTER_ID}'.format(END=endpoint_type, VDISK_ID=cli_volume_id,
+                                                                          OTHER_END=other_endpoint_type,
+                                                                          OTHER_VDISK_ID=other_cli_volume_id,
+                                                                          OTHER_CLUSTER_ID=other_system_id)
+        return self._get_lsrcrelationship(filter_value).as_list
+
+    def _get_rcrelationship(self, cli_volume_id, other_cli_volume_id, other_system_id):
+        rcrelationships = self._get_rcrelationships(cli_volume_id, other_cli_volume_id,
+                                                    other_system_id, as_master=True)
+        rcrelationships.extend(self._get_rcrelationships(cli_volume_id, other_cli_volume_id,
+                                                         other_system_id, as_master=False))
+        if len(rcrelationships) != 1:
+            logger.warning('found {0} rcrelationships for volume id {1} '
+                           'with volume id {2} of system {3}'.format(len(rcrelationships),
+                                                                     cli_volume_id,
+                                                                     other_cli_volume_id,
+                                                                     other_system_id))
+            return None
```

Comment:
why only a warning?

Reply:
should I change it to `error` and raise an exception?
in such scenario it will fail afterwards anyway

Reply:
I'm not sure, what is the actual meaning of getting here? what is the scenario?

Reply:
e.g. two identical rcrelationships but with opposite directions (shouldn't happen of course)

Reply:
so its a bug it should raise an exception IMO..

Reply:
I agree, it would be better to raise an exception here

Reply:
will be done in CSI-3394